repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,380 @@
+
+    new URL(portUrl.getProtocol, hostnameUrl, portUrl.getNumber, path)
+  }
+
+  def doDispatch(servletRequest: HttpServletRequestWrapper,
+                 servletResponse: HttpServletResponse,
+                 locationUrl: URL,
+                 uri: String,
+                 targetContext: ServletContext,
+                 routeDest: RouteDestination,
+                 dispatcher: RequestDispatcher,
+                 destination: Destination): Unit = {
+    servletRequest.setScheme(locationUrl.getProtocol)
+    servletRequest.setServerName(locationUrl.getHost)
+    servletRequest.setServerPort(locationUrl.getPort)
+    servletRequest.setRequestURI(locationUrl.getPath)
+
+    setVia(servletRequest)
+    setXForwardedFor(servletRequest)
+    logger.debug("Attempting to route to: {}", locationUrl.getPath)
+    logger.debug("  Using dispatcher for: {}", uri)
+    logger.debug("           Request URL: {}", servletRequest.getRequestURL)
+    logger.debug("           Request URI: {}", servletRequest.getRequestURI)
+    logger.debug("          Context path: {}", targetContext.getContextPath)
+    val startTime = System.currentTimeMillis
+    Try {
+      reportingService.incrementRequestCount(routeDest.getDestinationId)
+      dispatcher.forward(servletRequest, servletResponse)
+      // track response code for endpoint & across all endpoints
+      val endpoint = getEndpoint(destination, locationUrl)
+      val servletResponseStatus = servletResponse.getStatus
+      metricsService.foreach { metSer =>
+        markResponseCodeHelper(metSer, servletResponseStatus, endpoint)
+        markResponseCodeHelper(metSer, servletResponseStatus, AllEndpoints)
+        markRequestTimeoutHelper(metSer, servletResponseStatus, endpoint)
+        markRequestTimeoutHelper(metSer, servletResponseStatus, AllEndpoints)
+      }
+      val stopTime = System.currentTimeMillis
+      reportingService.recordServiceResponse(routeDest.getDestinationId, servletResponseStatus, stopTime - startTime)
+      fixLocationHeader(
+        servletRequest.getRequest.asInstanceOf[HttpServletRequest],
+        servletResponse,
+        routeDest,
+        locationUrl.getPath,
+        destination.getRootPath)
+    } recover {
+      case e: IOException if e.getCause.isInstanceOf[ReadLimitReachedException] =>
+        logger.error("Error reading request content", e)
+        trySendError(servletResponse, SC_REQUEST_ENTITY_TOO_LARGE, "Error reading request content")
+      case e =>
+        logger.error("Error communicating with {}", locationUrl.getPath, e)
+        servletResponse.setStatus(SC_SERVICE_UNAVAILABLE)
+    }
+  }
+
+  def setVia(request: HttpServletRequestWrapper): Unit = {
+    val builder = new StringBuilder
+    builder.append(HttpServletRequestUtil.getProtocolVersion(request))
+    val requestVia = containerConfigurationService.getRequestVia
+    builder.append(" ")
+    if (requestVia.isPresent) builder.append(requestVia.get)
```

Comment:
This line doesn't do anything? It appends to the `StringBuilder`, but then the method returns, and the `StringBuilder` is GC'd without doing anything.

Reply:
Yep. Another auto convert that failed.
Updated.